### PR TITLE
dev-libs/igraph: version bump to 0.9.0

### PR DESCRIPTION
**Pull request by epsilon-0**
moves to cmake based build system

Package-Manager: Portage-3.0.14, Repoman-3.0.2

### CODE REVIEW

File: dev-libs/igraph/igraph-0.9.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,52 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+CMAKE_MAKEFILE_GENERATOR="emake"
```
Comment (SoapGentoo) [inline dev-libs/igraph/igraph-0.9.0.ebuild:6]: why is emake required?

File: dev-libs/igraph/igraph-0.9.0.ebuild [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+inherit cmake
+
+DESCRIPTION="Creating and manipulating undirected and directed graphs"
+HOMEPAGE="http://www.igraph.org/"
+SRC_URI="https://github.com/igraph/igraph/releases/download/${PV}/${P}.tar.gz"
+
+LICENSE="GPL-2"
+SLOT="0/0"
```
Comment (SoapGentoo) [inline dev-libs/igraph/igraph-0.9.0.ebuild:14]: so the SONAME is `0`?

File: dev-libs/igraph/files/igraph-0.9.0-cmakedirs.patch [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+--- a/src/CMakeLists.txt
++++ b/src/CMakeLists.txt
+@@ -440,8 +440,8 @@ generate_export_header(igraph
+ include(GNUInstallDirs)
+ install(
+   TARGETS igraph
+-  RUNTIME DESTINATION bin
+-  LIBRARY DESTINATION lib
++  RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
++  LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
```
Comment (SoapGentoo) [inline dev-libs/igraph/files/igraph-0.9.0-cmakedirs.patch:12]: has this been submitted upstream?

File: dev-libs/igraph/igraph-0.9.0.ebuild [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+		-DIGRAPH_ENABLE_TLS=$(usex threads)
+		-DBUILD_TESTING=$(usex test)
+	)
+	cmake_src_configure
+}
+
+src_test(){
+	cd "${BUILD_DIR}"
+	emake check
+}
```
Comment (SoapGentoo) [inline dev-libs/igraph/igraph-0.9.0.ebuild:52]: nope, wrong, `cmake_src_test check`. Does the default test phase function not work?

**Review by SoapGentoo — CHANGES_REQUESTED** [review]

File: dev-libs/igraph/files/igraph-0.9.0-cmakedirs.patch (hunk 3 above)
Comment (epsilon-0) [reply dev-libs/igraph/files/igraph-0.9.0-cmakedirs.patch:12]: Am going to send it, they are pretty receptive and chill.

File: dev-libs/igraph/igraph-0.9.0.ebuild (hunk 4 above)
Comment (epsilon-0) [reply dev-libs/igraph/igraph-0.9.0.ebuild:52]: Yes, the default doesn't work, is also why I had to use `emake` as generator.
What you mentioned here, `cmake_src_test check`, also does not work.

File: dev-libs/igraph/igraph-0.9.0.ebuild (hunk 2 above)
Comment (epsilon-0) [reply dev-libs/igraph/igraph-0.9.0.ebuild:14]: Yes, the library name installed is `libigraph.so.0`

File: dev-libs/igraph/igraph-0.9.0.ebuild (hunk 1 above)
Comment (epsilon-0) [reply dev-libs/igraph/igraph-0.9.0.ebuild:6]: For the testing phase, as the default testing functions don't work.

File: dev-libs/igraph/igraph-0.9.0.ebuild (hunk 4 above)
Comment (epsilon-0) [reply dev-libs/igraph/igraph-0.9.0.ebuild:53]: Using `cmake_src_compile check` in src_test seems to work, would that be more acceptable?

Comment (SoapGentoo) [reply dev-libs/igraph/igraph-0.9.0.ebuild:53]: yes, use that please

Comment (SoapGentoo) [reply dev-libs/igraph/igraph-0.9.0.ebuild:53]: and work with upstream on just using CTest so we don't have to override it

File: dev-libs/igraph/igraph-0.9.0.ebuild (hunk 1 above)
Comment (epsilon-0) [reply dev-libs/igraph/igraph-0.9.0.ebuild]: removed

File: dev-libs/igraph/igraph-0.9.0.ebuild (hunk 4 above)
Comment (epsilon-0) [reply dev-libs/igraph/igraph-0.9.0.ebuild:51]: done

File: dev-libs/igraph/files/igraph-0.9.0-cmakedirs.patch [hunk 5]
```diff
@@ -0,0 +1,15 @@
+diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
+index 723d06d..bb6e041 100644
```
Comment (SoapGentoo) [inline dev-libs/igraph/files/igraph-0.9.0-cmakedirs.patch:2]: clean out the git metadata

File: dev-libs/igraph/igraph-0.9.0.ebuild [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+		-DIGRAPH_USE_INTERNAL_GLPK=OFF
+		-DIGRAPH_USE_INTERNAL_GMP=OFF
+		-DIGRAPH_USE_INTERNAL_LAPACK=OFF
+		-DIGRAPH_ENABLE_TLS=$(usex threads)
+		-DBUILD_TESTING=$(usex test)
+	)
+	cmake_src_configure
+}
+
+src_test(){
```
Comment (SoapGentoo) [inline dev-libs/igraph/igraph-0.9.0.ebuild:49]: `src_test() {`

**Review by SoapGentoo — CHANGES_REQUESTED** [review]

File: dev-libs/igraph/files/igraph-0.9.0-cmakedirs.patch (hunk 5 above)
Comment (epsilon-0) [reply dev-libs/igraph/files/igraph-0.9.0-cmakedirs.patch]: done

File: dev-libs/igraph/igraph-0.9.0.ebuild (hunk 6 above)
Comment (epsilon-0) [reply dev-libs/igraph/igraph-0.9.0.ebuild]: done

File: dev-libs/igraph/files/igraph-0.9.0-cmakedirs.patch (hunk 3 above)
Comment (szhorvat) [reply dev-libs/igraph/files/igraph-0.9.0-cmakedirs.patch:10]: igraph developer here. This is fixed in 0.9.1 now, and do let us know if there are any other problems. :-)

File: dev-libs/igraph/igraph-0.9.0.ebuild (hunk 4 above)
Comment (ntamas) [reply dev-libs/igraph/igraph-0.9.0.ebuild:51]: fyi, upstream has recently committed a [patch](https://github.com/igraph/igraph/commit/d829842c97cba752c4f1999a832a29ec2c174fbf) that makes it possible to use `make test` similarly to `make check`

Comment (szhorvat) [reply dev-libs/igraph/igraph-0.9.0.ebuild:51]: To be clear, this is not yet in 0.9.1, but it will be in 0.9.2.

**Pull request closed without merging**